### DISCUSSION TITLE
Starfield, a lovely Perlin Noise animation

### DISCUSSION THREAD

**Discussion by HaLo2FrEeEk** [rocksdanister/lively#1521] [Show and tell]
I've been working on this on and off for a while. A recent issue pulled me back into it and I think I've finally polished it up enough to put out there.

I love the smoothness of Perlin Noise, and these "stars" love to follow the flow as the noise field morphs over time. If you want you can also freeze the noise for a fun effect. It works perfectly across multiple monitors for that nice seamless look.

![](https://user-images.githubusercontent.com/2855829/210965892-244d012b-b91e-4f28-a4a4-ba153aba18c2.gif)

**Download**
[Starfield.zip](https://github.com/HaLo2FrEeEk/Starfield-wallpaper-for-Lively)

**Customizable options:**
Count
Speed
Size
Noise field resolution
Color
Rainbow mode!
Freeze mode

And a "bonus" fps counter option, for getting the most out of your Starfield :)

**Comment by rocksdanister** [comment]
Looks great, reminds me of Drift screensaver from Apple.

**Reply by HaLo2FrEeEk** [reply]
Oooh, I've never seen that before but yeah, I see the resemblance! It looks like Drift uses a grid of static dots that move and draw a line back to their origin, change color and size, based on an underlying perlin noise field. I like it :)

**Comment by naeq** [comment]
@HaLo2FrEeEk Really great work! Is there any way to change the background color? I need perfect black

**Reply by naeq** [reply]
got it! 
![image](https://user-images.githubusercontent.com/6697312/212399395-e92973df-b7dc-4218-a803-0db335705669.png)

**Reply by HaLo2FrEeEk** [reply]
Excellent! How did you end up doing it? Fade variable to 0? I could add that as a customization option, I did want to allow changing of the background anyway.

**Reply by HaLo2FrEeEk** [reply]
The problem I had with perfect black (and the reason I didn't try to do it) was the way the particles "fade", leaving a permanent trail, eventually covering the whole screen and ending up with a grayish. I added the loop in setup that draws the selected particle color, then draws the background like 100 times to get around the annoying trail issue without affecting performance.

If you found a way to do it I would love to hear it!
